src/pipecat/processors/user_idle_processor.py

```diff
@@ -32,28 +33,34 @@ def __init__(
         **kwargs,
     ):
         super().__init__(**kwargs)
-
+        self._running = True
         self._callback = callback
         self._timeout = timeout
-
         self._interrupted = False
-
         self._create_idle_task()
 
     async def process_frame(self, frame: Frame, direction: FrameDirection):
         await super().process_frame(frame, direction)
 
+        # Check for end frames before processing
+        if isinstance(frame, (EndFrame, EndTaskFrame)):
```

Comment:
No need to check for `EndTaskFrame`. `EndTaskFrame` will generate an `EndFrame`.